src/clientmetricreport/DefaultClientMetricReport.ts

```diff
@@ -418,6 +525,33 @@ export default class DefaultClientMetricReport implements ClientMetricReport {
     return metric;
   }
 
+  getObservableVideoUpstreamMetrics(): { [id: string]: {} } {
+    return this.getObservableVideoMetrics(Direction.UPSTREAM);
+  }
+
+  getObservableVideoDownstreamMetrics(): { [id: string]: {} } {
+    return this.getObservableVideoMetrics(Direction.DOWNSTREAM);
+  }
+
+  getObservableVideoMetrics(direction: number): { [id: string]: {} } {
+    const mediaStreamMetrics: { [id: string]: {} } = {};
+    for (const ssrc in this.streamMetricReports) {
+      if (
+        this.streamMetricReports[ssrc].direction === direction &&
+        this.streamMetricReports[ssrc].mediaType === MediaType.VIDEO
+      ) {
+        const metric: { [id: string]: number } = {};
+        for (const metricName in this.observableVideoMetricSpec) {
+          if (this.observableVideoMetricSpec[metricName].dir === direction) {
+            metric[metricName] = this.getObservableVideoMetricValue(metricName, Number(ssrc));
+          }
+        }
+        mediaStreamMetrics[ssrc] = metric;
```

Comment:
I dont think we should index it under ssrc. From the builders, there is no way to map from attendeeId or tileid to ssrc right?
How do builders know which one refer to which video tile or attendee?

Reply:
@XHatan What do you think? Can we use streamId?

Reply:
Seems we can't get streamId for upstreams with this code: 
`this.videoStreamIndex.streamIdForSSRC(Number(rawMetricReport.ssrc));`
Is this correct? @XHatan

Reply:
Isn't streamId available in streamMetricReport.streamId?

Reply:
I tested in the streamMetricReport and only see streamId for the downstream video metrics. Other streams do not have streamId. This is the line of code to add streamIds to the streamMetricReport: https://github.com/aws/amazon-chime-sdk-js/blob/master/src/statscollector/DefaultStatsCollector.ts#L250-L254